src/Microsoft.AspNet.Hosting/Internal/WebApplication.cs

```diff
@@ -98,14 +127,14 @@ public virtual IApplication Start()
             _applicationLifetime.NotifyStarted();
             logger.Started();
 
-            return new Application(ApplicationServices, Server.Features, new Disposable(() =>
-            {
-                logger.Shutdown();
-                _applicationLifetime.StopApplication();
-                Server.Dispose();
-                _applicationLifetime.NotifyStopped();
-                (_applicationServices as IDisposable)?.Dispose();
-            }));
+            return new Disposable(() =>
```

Comment:
We said not to return just an IDisposable because we would want to add stuff. If you're just going to do IDisposable then make IWebApplication and return void here.